src/ratelimit.ts

```diff
@@ -129,7 +129,7 @@ export abstract class Ratelimit<TContext extends Context> {
    *  return "Yes"
    * ```
    */
-  public limit = async (identifier: string, req?: { geo?: Geo }, customRates?: { [key: string]: number }): Promise<RatelimitResponse> => {
+  public limit = async (identifier: string, req?: { geo?: Geo }, customRates?: number): Promise<RatelimitResponse> => {
```

Comment:
@ogzhanolguncu @sourabpramanik would it be better if `customRates` is inside `req` object?
If you do like that, then you won't have to specify `undefined` when calling limit function like in this example: `await ratelimit.limit(ip ?? "anonymous011", undefined, 60);`

Also maybe it should be `customRate` instead.

Reply:
I have already noticed that and since this is draft PR a lot of refactoring is going to happen. We need to see if these algorithm changes are accepted or not.

Reply:
Yeah, I agree with @off99555. Idea is pretty good let's move forward with your new changes @sourabpramanik.

Reply:
> Yeah, I agree with @off99555. Idea is pretty good let's move forward with your new changes @sourabpramanik.

Sweet